pkgs/applications/networking/StevenBlack-hosts/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenvNoCC, fetchFromGitHub, python3, lib
+, flags ? {} ,allowList ? ""}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "StevenBlack-hosts";
+  version = "2.5.52";
+  src = fetchFromGitHub {
+    owner = "StevenBlack";
+    repo = "hosts";
+    rev = version;
+    sha256 ="01 rijzfmvwfcirz94h74bx15maqdn9m01g7q197nvzxmr1fk60yd";
```

Comment:
Why is here a space?

Reply:
The space character issue aside, this makes me think I should be publishing the hash for each hosts file I generate to ensure the integrity of the chain of custody.

@SuperSandro2000 who generates the hash currently?  It's most unusual to have a hash created downstream from its origin.￼  

Am I missing context here?

I am happy to generate the hashes in the future...￼

Reply:
I have generated the hash, simply by running `nix-prefetch fetchFromGitHub …`.
I expect the update scripts to update said hash on version update.

It's pratically required for nixpkgs to include the hash of the source, as one of its objectives is to ensure reproducibility of the package builds.

Reply:
Thanks @pasqui23.

Another observation.  The latest release of hosts is [3.5.2](https://github.com/StevenBlack/hosts/releases/tag/3.5.2).  Release [2.5.52](https://github.com/StevenBlack/hosts/releases/tag/2.5.52), referenced here, is from Feb 9, 2020.  More than a year ago.  We release between 1 and 3 times per week, depending on what comes in.

Am I missing context here?

Reply:
Adding an hash per host file would not be used by nixpkgs, as the hash already check the ccontents of *every* host file in the repo.

Reply:
> The space character issue aside, this makes me think I should be publishing the hash for each hosts file I generate to ensure the integrity of the chain of custody.

We do not verify such hashes normally and because of some normalization done fetchFromGitHub will most likely return different hashes anyway.

> who generates the hash currently? It's most unusual to have a hash created downstream from its origin.

That is totally normal for nix and hashes are verified by the CI and everyone who tries to build the packages locally because the hashes are not already in the binary cache.

> Another observation. The latest release of hosts is 3.5.2. Release 2.5.52, referenced here, is from Feb 9, 2020. More than a year ago. We release between 1 and 3 times per week, depending on what comes in.

The review process took some time so we will probably need to update it right after merging.